detection_rules/rule.py

```diff
@@ -714,6 +714,21 @@ def interval_ratio(self) -> Optional[float]:
             return interval / self.max_span
 
 
+@dataclass(frozen=True)
+class ESQLRuleData(BaseRuleData):
```

Comment:
We actually should be inheriting `QueryRuleData` instead. `AlertSuppression` is already scoped solely to `query` rules atm, which makes it fully compatible and consistent with our inheritance strategy

Reply:
ah, good to know. We can easily override it in the same way we did for suppression. Will add